microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,325 @@
+                    "expect",
+                    "dh-autoreconf",
+                    "gdb",
+                    "libnuma-dev",
+                    "quota",
+                    "genisoimage",
+                    "db-util",
+                    "unzip",
+                    "exfat-utils",
+                ]
+            )
+        elif isinstance(self.node.os, Suse):
+            self.node.os.install_packages(
+                [
+                    "ntp",
+                    "git-core",
+                    "db48-utils",
+                    "libaio-devel",
+                    "libattr1",
+                    "libcap-progs",
+                    "libdb-4_8",
+                    "perl-BerkeleyDB",
+                ]
+            )
+        elif isinstance(self.node.os, CBLMariner):
+            self.node.os.install_packages(
+                [
+                    "kernel-headers",
+                    "binutils",
+                    "glibc-devel",
+                    "zlib-devel",
+                ]
+            )
+        else:
+            raise LisaException(f"{self.node.os} is not supported")
+
+        # Some CPU time is assigned to set real-time scheduler and it affects
+        # all cgroup test cases. The values for rt_period_us(1000000us or 1s)
+        # and rt_runtime_us (950000us or 0.95s). This gives 0.05s to be used
+        # by non-RT tasks.
+        if self.node.shell.exists(
+            PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us")
+        ):
+            runtime_us = self.node.tools[Cat].read(
+                "/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us",
+                force_run=True,
+                sudo=True,
+            )
+            runtime_us_int = int(runtime_us)
+            if runtime_us_int == 0:
+                self.node.tools[Echo].write_to_file(
+                    "1000000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/cpu.rt_period_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "950000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/cpu.rt_runtime_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "1000000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_period_us"),
+                    sudo=True,
+                )
+                self.node.tools[Echo].write_to_file(
+                    "950000",
+                    PurePosixPath("/sys/fs/cgroup/cpu/user.slice/cpu.rt_runtime_us"),
+                    sudo=True,
+                )
+
+        # Minimum 4M swap space is needed by some mmp test
+        if self.node.tools[Free].get_swap_size() < 4:
+            self.node.execute("dd if=/dev/zero of=/tmp/swap bs=1M count=1024")
+            self.node.tools[MkSwap].run("/tmp/swap")
+            self.node.tools[SwapOn].run("/tmp/swap")
```

Comment:
make sure to restore them back to original status, it may have impact for other cases.

Reply:
I will add the code for removing swap but I think there might be a bigger issue. I found following warning on LTP github repo:
```
Don't run them on production systems. Growfiles, doio, and iogen in particular stress the I/O capabilities of systems and while they should not cause problems on properly functioning systems, they are intended to find (or cause) problems.
```

I think we should mark the nodes as dirty which run ltp tests. Any bugs in the ltp which don't lead to clean exit of tests could cause problems with lisa testing. @squirrelsc @LiliDeng suggestions?

Reply:
> I will add the code for removing swap but I think there might be a bigger issue. I found following warning on LTP github repo:
> 
> ```
> Don't run them on production systems. Growfiles, doio, and iogen in particular stress the I/O capabilities of systems and while they should not cause problems on properly functioning systems, they are intended to find (or cause) problems.
> ```
> 
> I think we should mark the nodes as dirty which run ltp tests. Any bugs in the ltp which don't lead to clean exit of tests could cause problems with lisa testing. @squirrelsc @LiliDeng suggestions?

If it's not recoverable, mark the node to dirty.